src/components/event_merchant/login.vue

```diff
@@ -82,31 +84,114 @@ export default {
             email:"",
             password:""
           },
+
       }
   },
-
+    
+    computed:{
+    ...mapGetters({errors:'returnMessage',success:'returnData', auth:'isAuthenticated'}),
+  },
+  
   methods: {
-    ...mapActions(["LogIn"]),
+    ...mapActions(["fbLogin","LogIn",'loadFacebookSDK','initFacebook','getUserDetails']),
+    
+    // this function will make the token available
     async submit() {
-        const User={
-            email: this.form.email,
-            password: this.form.password
+        
+        try {
+            const User={
+                email: this.form.email,
+                password: this.form.password
+            }
+            await this.LogIn(User);
+        } catch (error) {
+            console.log(error);
         }
-      try {
-        await this.LogIn(User);
-        if(this.success){
-            this.$router.go({name:"VendorProfile"});
+      
+    },
+    routePath(path){
+        return this.$router.go({name:path});
+    },
+    // While this function make use of the token.
+    async getSubmit(){
+      await this.submit()
+      const token = localStorage.getItem('token');//get token from LS
+        try {
+                  
+            const decodeToken = jwt_decode(token);//decode the token
+            localStorage.setItem('userId',decodeToken['http://schemas.xmlsoap.org/ws/2005/05/identity/claims/sid'][1]);
+            const role = localStorage.setItem('role',decodeToken.role);//store role to Localstorage
+
+            const userCre={
+                userId: localStorage.getItem('userId'),
+                bearerToken: localStorage.getItem('token'),
+            };
+            await this.getUserDetails(userCre);
+
+            if(this.success === "success" && role === 'User'){
+                this.routePath("UserProfile")
+                //this.$router.go({path:"/user_profile"});
+            }else{
```

Comment:
This logic should also check for Vendor roles as well and redirect to login if the role doesn't match any of User or Vendor, It always redirects to Vendor's profile page if the  Role is not User.

Reply:
Resolved